department-of-veterans-affairs/vets-website
# Update UI for DSLogon and MHV users

## Summary

- 2 pages of Profile application are updated
  - Contact Information is updated to conditionally include a 'Sign In Email' section only if the account is ID.me or Login.gov, since they use emails as their login credential. If the user is MHV or DSLogon, then that section is now omitted as they use a username for their login credential, and it was misleading to show that section. 
- Account Security has one minor text adjustment to update 'email address' to 'information' to make it work for any account type / login credential type.
- Profile team owns this work and will maintain it.

To test:
- [Run Profile FE in watch mode](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/profile/engineering-docs/frontend-getting-started.md#start-webpack-to-compile-and-watch-the-profile-code-and-recompile-after-changes-to-the-code-are-saved)
- [Run mock server](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/identity-personalization/profile/engineering-docs/frontend-getting-started.md#start-a-mock-server-that-will-create-responses-for-various-endpoints-that-are-needed-to-display-the-profile)
- Update the mock user response [here](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/personalization/profile/mocks/endpoints/user.js#L1447) to instead use `allMockResponses.mvhUser`
- View Contact Info and Account Security page to verify changes

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/47070

## Testing done

- No unit tests were present for the `EmailInformationSection` component and so I have added tests to cover this new conditional logic as well as a main test for the Contact email address section.

## Screenshots

Contact Information Page
| Before | After |
| ------ | ----- |
| ![Screenshot 2023-06-01 at 1 02 12 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/d7757758-d1e5-4e4b-b693-dd2e9abaf2c2) | ![Screenshot 2023-06-01 at 12 54 41 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/971345f2-f350-4f23-98d0-15a68ecc26c1) |

Account Security Page
| Before | After |
| ------ | ----- |
| ![Screenshot 2023-06-01 at 12 58 15 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/2dc10dba-943d-4c6c-a2d7-a0909e963d36) | ![Screenshot 2023-06-01 at 12 59 09 PM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/e7599f4d-a688-4f6f-998d-dc650f0ad42b) |

## What areas of the site does it impact?

Profile Application - Contact Information and Account Security pages

## Acceptance criteria

- [x] Ui for contact page updated
- [x] UI for account security updated

### Quality Assurance & Testing


### Error Handling
